src/mito2/src/read/scan_util.rs

```diff
@@ -14,23 +14,301 @@
+        Self {
+            prepare_scan_cost: MetricBuilder::new(metrics_set)
+                .subset_time("prepare_scan_cost", partition),
+            build_parts_cost: MetricBuilder::new(metrics_set)
+                .subset_time("build_parts_cost", partition),
+            build_reader_cost: MetricBuilder::new(metrics_set)
+                .subset_time("build_reader_cost", partition),
+            scan_cost: MetricBuilder::new(metrics_set).subset_time("scan_cost", partition),
+            convert_cost: MetricBuilder::new(metrics_set).subset_time("convert_cost", partition),
+            yield_cost: MetricBuilder::new(metrics_set).subset_time("yield_cost", partition),
+            total_cost: MetricBuilder::new(metrics_set).subset_time("total_cost", partition),
+            num_rows: MetricBuilder::new(metrics_set).counter("num_rows", partition),
+            num_batches: MetricBuilder::new(metrics_set).counter("num_batches", partition),
+            num_mem_ranges: MetricBuilder::new(metrics_set).counter("num_mem_ranges", partition),
+            num_file_ranges: MetricBuilder::new(metrics_set).counter("num_file_ranges", partition),
+            rg_total: MetricBuilder::new(metrics_set).counter("rg_total", partition),
+            rg_fulltext_filtered: MetricBuilder::new(metrics_set)
+                .counter("rg_fulltext_filtered", partition),
+            rg_inverted_filtered: MetricBuilder::new(metrics_set)
+                .counter("rg_inverted_filtered", partition),
+            rg_minmax_filtered: MetricBuilder::new(metrics_set)
+                .counter("rg_minmax_filtered", partition),
+            rg_bloom_filtered: MetricBuilder::new(metrics_set)
+                .counter("rg_bloom_filtered", partition),
+            rows_before_filter: MetricBuilder::new(metrics_set)
+                .counter("rows_before_filter", partition),
+            rows_fulltext_filtered: MetricBuilder::new(metrics_set)
+                .counter("rows_fulltext_filtered", partition),
+            rows_inverted_filtered: MetricBuilder::new(metrics_set)
+                .counter("rows_inverted_filtered", partition),
+            rows_bloom_filtered: MetricBuilder::new(metrics_set)
+                .counter("rows_bloom_filtered", partition),
+            rows_precise_filtered: MetricBuilder::new(metrics_set)
+                .counter("rows_precise_filtered", partition),
+            num_sst_record_batches: MetricBuilder::new(metrics_set)
+                .counter("num_sst_record_batches", partition),
+            num_sst_batches: MetricBuilder::new(metrics_set).counter("num_sst_batches", partition),
+            num_sst_rows: MetricBuilder::new(metrics_set).counter("num_sst_rows", partition),
+            first_poll: MetricBuilder::new(metrics_set).subset_time("first_poll", partition),
+        }
+    }
+
+    /// Merges the local scanner metrics.
+    fn merge_scanner_metrics(&self, other: &ScannerMetrics) {
```

Comment:
Suggest to use struct destruction here to prevent from forgetting merging some fields if any more fields were added in the future. For example,
```rust
struct ScannerMetrics {
  old_fields: i32,
  new_field: String,
}

let ScannerMetrics {
  old_fields,
  // if "new_field" is missing here, compiler will emit an error, and you will know
} = other;
```